crates/rspack_core/src/tree_shaking/optimizer.rs

```diff
@@ -554,6 +554,7 @@ impl<'a> CodeSizeOptimizer<'a> {
     }
     // visited_module.remove(&cur);
 
+    // TODO: this code can be slightly more concise
```

Comment:
I am not sure if you mean to use a boolean instead, the reason why I use a list here is `Vec` is debug-friendly. If not, could you provide more details?

Reply:
We can use 

```
match side_effects_map.get_mut() {
  Some(SideEffect::Configuration(_)) |   None => vec![],
  Some(SideEffect::Analyze(_)) => {
    xxx
  }
}
```

Reply:
LGTM

Reply:
I can make the change in following pr. Keep this change relevant only to tree-shaking